src/html5/parser.rs

```diff
@@ -187,12 +187,14 @@ pub struct Html5Parser<'stream> {
     parser_pause_flag: bool,
     /// Keeps the position of where any document.write() should be inserted when running a script
     insertion_point: Option<usize>,
-    // ignore when next token is LF
+    /// ignore when next token is LF
     ignore_lf: bool,
-    // Sometimes tokens needs to be split up (and it seems the tokenizer cannot do this?)
+    /// Sometimes tokens needs to be split up (and it seems the tokenizer cannot do this?)
     token_queue: Vec<Token>,
-    // When true, the parser is finished and should not consume more tokens (there aren't any)
+    /// When true, the parser is finished and should not consume more tokens (there aren't any)
     parser_finished: bool,
+    /// Context node for fragment parsing
+    context: Option<NodeId>,
```

Comment:
Is there a way to make this non-option?
1. In the fragment case, you have some random NodeId
2. In the non-fragment case, you have the root id

Reply:
in theory yes.. The `context` node is only set when `is_fragment_case` is `true`.  This can cause an issue when `is_fragment_case` is set, but the `context` node isn't (we use `expect()` to fetch the context in fragment cases). 

However, the only place the `context` and `is_fragment_case` are set, is set is in the `new_*` constructors.  We could remove `is_fragment_case` and have a function `is_fragment_case()` that checkx if `context` is `Some()`, but we couldn't set the `context` to root in non-fragment cases.

However, next iteration already changes this context node so I'll see what we can do there to make it less error-prone